core/domain/user_id_migration.py

```diff
@@ -182,16 +182,21 @@ def _replace_gae_ids(gae_ids):
         Raises:
             MissingUserException: UserSettingsModel with GAE ID doesn't exist.
         """
-        new_ids = []
-        for gae_id in gae_ids:
-            if gae_id == feconf.SYSTEM_COMMITTER_ID:
-                new_ids.append(feconf.SYSTEM_COMMITTER_ID)
-            else:
-                user_settings_model = (
-                    user_models.UserSettingsModel.get_by_gae_id(gae_id))
-                if not user_settings_model:
+        non_admin_gae_ids = [
+            gae_id for gae_id in gae_ids
+            if gae_id != feconf.SYSTEM_COMMITTER_ID]
+        user_setting_models = user_models.UserSettingsModel.query(
```

Comment:
Ah -- I am not sure this is actually faster. Normally get_multi() is faster than N gets but that's only if you are getting by the model ID/key. I believe query() and IN are implemented as an OR with N separate queries.

Could you do a manual test to compare the time for the two approaches?

Reply:
After more thorough testing I find out that this approach didn't work anyway. Moved back to the previous state.